packages/js/src/utils/minimal-publish-script.ts

```diff
@@ -0,0 +1,49 @@
+import type { Tree } from '@nrwl/devkit';
+
+const publishScriptContent = `
+/**
+* This is a minimal script to publish your package to "npm".
+* This is meant to be used as-is or customize as you see fit.
+* 
+* This script is executed on "dist/path/to/library" as "cwd" by default.
+*
+* You might need to authenticate with NPM before running this script.
+*/
+
+import { execSync } from 'child_process';
+import { readFileSync, writeFileSync } from 'fs';
+import * as chalk from 'chalk';
+
+// Executing publish script: node path/to/publish.mjs {name} --version {version} --tag {tag}
+// Default "tag" to "next" so we won't publish the "latest" tag by accident.  
+const [, , name, version, tag = 'next'] = process.argv;
+
+// A simple SemVer validation to validate the version 
+const validVersion = /^\\d+\\.\\d+\\.\\d(-\\w+\\.\\d+)?/;
+if (!version || !validVersion.test(version)) {
+  console.error(chalk.bold.red(\`No version provided or version did not match Semantic Versioning, expected: #.#.#-tag, got \${version}\`));
```

Comment:
messed up my understanding of the regex. this would be a better error message to explain what was expected
```suggestion
  console.error(chalk.bold.red(\`No version provided or version did not match Semantic Versioning, expected: #.#.#-tag.# or #.#.#, got \${version}\`));
```